dao/application_type_dao.go

```diff
@@ -30,8 +30,11 @@ func (a *ApplicationTypeDaoImpl) SubCollectionList(primaryCollection interface{}
 
 	// limiting + running the actual query.
 	result := query.Limit(limit).Offset(offset).Find(&applicationTypes)
+	if result.Error != nil {
+		return nil, 0, util.NewErrBadRequest(result.Error.Error())
+	}
```

Comment:
I would suggest the following:

```diff
- result := query.Limit(limit).Offset(offset).Find(&applicationTypes)
+ err := query.Limit(limit).Offset(offset).Find(&applicationTypes).Error
```

This way you don't have to chain the `result.Error.Error()` members and functions.